### ISSUE TITLE
[In progress] PixelCNN [Pixel Recurrent Neural Networks]

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#24]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1601.06759
- 公開日時：2016/01/25
- 被引用数（記事作成時点）：751 件
- 実装コード：

## 1. どんなもの？
- 画像の２次元の空間次元方向に沿って、RNN で時系列的に次のピクセル分布を予想することを繰り返すことで、画像生成を行う自己回帰型の生成モデル [autoregressive models]。

## 2. 先行研究と比べてどこがすごいの？
- 自己回帰型の生成モデルにおいて、RNN と residual connections のアーキテクチャを採用することにより、既存の SOTA 手法よりも、鮮明で多様な画像生成を可能にしている。

## 3. 技術や手法の"キモ"はどこにある？
- 画像の自己回帰モデルの基本
    ![image](https://user-images.githubusercontent.com/25688193/66124317-07c16400-e61f-11e9-9970-ad3057f99434.png)
	画像の自己回帰モデルにおいては、上図のように、画像の左上のピクセルを開始時点、右下のピクセルを終了時点として、上行から下行へ、各行で左から右へ移動しながら次のピクセルの条件付き確率分布を予想することを逐次繰り返すことで画像生成を行う。最終的な画像全体の確率分布は、以下の式のように、これら各ピクセルの条件付き確率の積（＝同時確率分布）で得られる。
    ![image](https://user-images.githubusercontent.com/25688193/66124338-10b23580-e61f-11e9-8adc-2ca7cb0585a1.png)
	又、各ピクセルのRBG 値は、R → G → B の順に予想し、各々のピクセルの値は、これら条件付き確率の積（＝同時確率分布）で得られる。
    ![image](https://user-images.githubusercontent.com/25688193/66124379-2a537d00-e61f-11e9-9607-ecdbe25dbcd2.png)
	本手法では、このような画像の自己回帰モデルに対して、その長期依存性をうまく捉えるために、RNN の LSTM でモデル化する。
	更に、条件付き確率分布を softmax 層での多項分布で近似することで、離散的な確率分布でモデル化する。これにより、連続的な確率分布でモデル化するよりも、学習しやすくなるという利点が得られる。

- PixelRNN
    	PixelRNN は以下のような Row LSTM と Diagonal BiLSTM の２種類のタイプの LSTM から構成される。

    - Row LSTM
        ![image](https://user-images.githubusercontent.com/25688193/66124415-48b97880-e61f-11e9-82f0-0dd949d9ecf5.png)
		Row LSTM は、画像の各行全体で１次元の畳み込みを行う。
		この畳み込みを各行で繰り返すことで、上図のように三角形状に畳み込みが行われて情報が集約されることになる。
		
		より詳細には、2次元画像 x 全体に対して、LSTM の各ゲートを、一般的な LSTM と同様にして、以下の式に従って更新する。
		※ 下式での畳み込みカーネル K による畳み込みで、行全体に対しての１次元の畳み込みを行っている？
        ![image](https://user-images.githubusercontent.com/25688193/66124432-53740d80-e61f-11e9-928b-9b8da0af37ea.png)
        ![image](https://user-images.githubusercontent.com/25688193/66124493-828a7f00-e61f-11e9-8ffc-2e3b3c034e88.png)

    - Diagonal BiLSTM
        ![image](https://user-images.githubusercontent.com/25688193/66124444-5ec73900-e61f-11e9-8b57-08edbe2fe9e8.png)
        xxx

- PixelCNN
    ![image](https://user-images.githubusercontent.com/25688193/66124543-a1891100-e61f-11e9-809d-677bcc5ae2ae.png)
    xxx

## 4. どうやって有効だと検証した？
- xxx

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- xxx
